### ISSUE TITLE
Bluetooth audio on PipeWire not ready for production yet

### ISSUE THREAD

**Issue by magiruuvelvet** [magiruuvelvet/gentoo-portage-config#1]
Bluetooth audio is utterly broken with PipeWire when multiple streams are active or if streams are rapidly opened/closed, **no issues with PulseAudio**. USB Audio (FiiO M5) works perfectly fine and can handle the heat and never breaks when spamming streams or having multiple concurrent streams open.

```plain
wireplumber[112313]: (-49) client too slow! rate:1024/48000 pos:23148544 status:triggered
```

Investigation is ongoing. Experimenting with different quantum values and sample rates didn't lead to any meaningful results yet. My only guess is that the Bluetooth plugin in PipeWire does some weird buffering.

Kernel version: `5.15.119-gentoo-dist`

For reference: Bluetooth audio on PulseAudio works perfectly fine on my system. Spamming streams was never an issue there. Why does this matter? Applications which play notification sounds, or simply just having sneaky audio playback in web browsers while listening to music, breaks the playback on PipeWire and I need to restart it in the worst case, sometimes switching profiles fixes it. This is far from being useable in production right now.

Debugging: You can install PulseAudio and PipeWire at the same time by using my [media-sound/pulseaudio-daemon-noconflict](https://github.com/magiruuvelvet/gentoo-overlay/tree/master/media-sound/pulseaudio-daemon-noconflict) ebuild. **Only system mode is supported.**

**Comment by magiruuvelvet** [comment]
Interesting ticket upstream: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/3316

Not sure if this problem is related to this issue, since only ALSA devices are mentioned there (Bluetooth audio devices are listed as BlueZ, not ALSA, in PipeWire), but keep an eye on that one.

**Comment by magiruuvelvet** [comment]
**New discovery:** The Bluetooth audio playback only becomes unstable when you **mix streams of different sample rates** and the resampler kicks in. Which means if you have a music library with different sample rates (Hi-Res, Standard Res, etc.) you will most likely experience audio cut-outs and crashes in PipeWire during Bluetooth playback.

PipeWire on Bluetooth didn't crash a single time on me when there was a consistent sample rate during my testing.

**Comment by magiruuvelvet** [comment]
Possible solution found, trying version [`0.3.73`](https://gitlab.freedesktop.org/pipewire/pipewire/-/tags/0.3.73) with [these](https://github.com/magiruuvelvet/gentoo-portage-config/tree/41f52208d3dd2cc37560adbf1fce646d4d1125a6/patches/media-video/pipewire) patches applied on top.

**Comment by magiruuvelvet** [comment]
Fixed in [`0.3.74`](https://gitlab.freedesktop.org/pipewire/pipewire/-/tags/0.3.74). In the meantime, apply [this patchset](https://github.com/magiruuvelvet/gentoo-portage-config/tree/c6b55ed07a94fa218365266011f261159fe61250/patches/media-video/pipewire) to get the fixes early.
